nomad/node_pool_endpoint.go

```diff
@@ -383,3 +383,93 @@ func (n *NodePool) ListJobs(args *structs.NodePoolJobsRequest, reply *structs.No
 		}}
 	return n.srv.blockingRPC(&opts)
 }
+
+// ListNodes is used to retrieve a list of nodes for a give node pool. It
+// supports pagination and filtering.
+func (n *NodePool) ListNodes(args *structs.NodePoolNodesRequest, reply *structs.NodePoolNodesResponse) error {
+	authErr := n.srv.Authenticate(n.ctx, args)
+	if done, err := n.srv.forward("NodePool.ListNodes", args, args, reply); done {
+		return err
+	}
+	n.srv.MeasureRPCRate("node_pool", structs.RateMetricRead, args)
```

Comment:
I sort of regret that we have this metric split between read and list, but:

```suggestion
	n.srv.MeasureRPCRate("node_pool", structs.RateMetricList, args)
```

Reply:
Ah thanks. I thought this was like "I'm reading the node pool", but it does return a list, so I guess from a rate limiting point of view this is a more expensive endpoint. 

I fixed this is `ListJobs` as well 👍

Reply:
D'oh, I missed it in `ListJobs` :grinning: